CalibTracker/SiStripCommon/BuildFile.xml

```diff
@@ -4,6 +4,7 @@
 <use name="DataFormats/SiStripDetId"/>
 <use name="DataFormats/Scalers"/>
 <use name="DataFormats/OnlineMetaData"/>
+<use name="SimDataFormats/TrackingAnalysis"/>
```

Comment:
why did you move `SimDataFormats/TrackingAnalysis` from `CalibTracker/SiStripCommon/plugins/BuildFile.xml` to `CalibTracker/SiStripCommon/BuildFile.xml`?

Reply:
@silviodonato in an earlier iteration before rebasing I had moved some headers to the `interface` directory, and that required this change. For the final version I dropped that change, so shall I put `SimDataFormats/TrackingAnalysis` back to where it was?

Reply:
I've already seen a similar discussion in a @guitargeek's PR https://github.com/cms-sw/cmssw/pull/31533#discussion_r494166871

Rule 6.6 of https://cms-sw.github.io/cms_coding_rules.html suggests to have a dedicated `BuildFile.xml` for plugins.
I think this prevents to have cyclic dependencies, but I'm not 100% sure.

> shall I put SimDataFormats/TrackingAnalysis back to where it was?

it is not urgent, but you can include it in your next PR